### PR TITLE
Add fractionalDigits to BcpTicker; implement getTicker/getAllTickerts for Ethereum

### LINKED ISSUE #640: Migrate transaction .chainId and .signer into .creator
Convert `UnsignedTransaction.chainId` + `UnsignedTransaction.signer` into

```ts
interface UnsignedTransaction {
  creator: PublicIdentity;
  // ...
}
```

This consistes of two aspects:

* Merging `chainId` and `signer` into a combined object of type `PublicIdentity`
* Use the term "creator" as a better name when you think of multi signatures

### PR DESCRIPTION
**Pull request by webmaster128**
Closes #640

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +25,10 @@ Breaking changes
   `BcpAccount | undefined` to better represent the one-or-none result.
 * @iov/bcp-types: Remove `BcpValueNameQuery` from `.getAccount` and `.watchAccount`
   as we're migrating from wallet nicknames to username NFTs.
+* @iov/bcp-types: Convert `BcpQueryEnvelope` to `ReadonlyArray` in return
```
Comment (ethanfrey) [inline CHANGELOG.md]: :+1:

File: packages/iov-bns/src/decode.ts [hunk 2]
```diff
@@ -101,6 +101,7 @@ export function decodeToken(data: codecImpl.namecoin.IToken & Keyed): BcpTicker
   return {
     tokenTicker: Encoding.fromAscii(data._id) as TokenTicker,
     tokenName: ensure(data.name),
+    fractionalDigits: 9, // fixed for all weave tokens
```
Comment (ethanfrey) [inline packages/iov-bns/src/decode.ts]: Yes

File: packages/iov-ethereum/src/ethereumconnection.ts [hunk 3]
```diff
@@ -211,12 +210,18 @@ export class EthereumConnection implements BcpConnection {
...
   }
 
-  public getAllTickers(): Promise<BcpQueryEnvelope<BcpTicker>> {
-    throw new Error("Not implemented");
+  public async getAllTickers(): Promise<ReadonlyArray<BcpTicker>> {
+    return [
+      {
+        tokenTicker: constants.primaryTokenTicker,
+        tokenName: constants.primaryTokenName,
+        fractionalDigits: constants.primaryTokenFractionalDigits,
```
Comment (ethanfrey) [inline packages/iov-ethereum/src/ethereumconnection.ts]: Good for now, will have to revisit with erc20 support

**Comment by webmaster128** [comment]
For some reason this diff triggers an unrelated bug, which also exists on master: in "watches headers with same data as getBlockHeader" the producer stop implementation

```ts
      stop: () => {
        headersSubscription.unsubscribe();
        // see https://github.com/ethereum/go-ethereum/wiki/RPC-PUB-SUB#cancel-subscription
        const listener = getListenerById("watchBlockHeaders");
        this.socketSend(
          JSON.stringify({
            id: "watchBlockHeaders",
            jsonrpc: "2.0",
            method: "eth_unsubscribe",
            params: [listener!.subscription],
          }),
        );
      },
```

runs after socket disconnection since stop() is always called asynchonously in xstream. As far as I understand we must never throw and exception in Producer.stop. See also https://github.com/staltz/xstream/issues/269

**Pull request merged**